core/templates/cowdata.h

```diff
@@ -167,6 +167,10 @@ class CowData {
 
 public:
 	void operator=(const CowData<T> &p_from) { _ref(p_from); }
+	void operator=(CowData<T> &&p_from) {
+		_ptr = p_from._ptr;
```

Comment:
If you are valid, you need to unref yourself first. Or previous memory will never be freed.

Reply:
Oh yeah, right...

Reply:
I switched the implementation to use `std::swap`, that should ensure `_ptr` is deallocated when `p_from` is destructed.